internal/olm/client/client.go

```diff
@@ -214,7 +216,53 @@ func (c Client) DoCSVWait(ctx context.Context, key types.NamespacedName) error {
 		return curPhase == olmapiv1alpha1.CSVPhaseSucceeded, nil
 	}
 
-	return wait.PollImmediateUntil(time.Second, csvPhaseSucceeded, ctx.Done())
+	err := wait.PollImmediateUntil(time.Second, csvPhaseSucceeded, ctx.Done())
+	if err != nil && errors.Is(err, context.DeadlineExceeded) {
+		depCheckErr := c.getDeploymentStatus(ctx, key, csv)
+		if depCheckErr != nil {
+			return fmt.Errorf("failed to run operator: %v %v", err, depCheckErr)
+		}
+	}
+
+	return err
+}
+
+// Get Deployment status
+
+func (c Client) getDeploymentStatus(ctx context.Context, key types.NamespacedName, csv olmapiv1alpha1.ClusterServiceVersion) error {
+	dep := &appsv1.Deployment{}
+	var depName string
+	for _, depSpec := range csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs {
+		if depSpec.Spec.Selector.MatchLabels["control-plane"] == "controller-manager" {
+			depName = depSpec.Name
+		}
+	}
+	depKey := types.NamespacedName{
+		Namespace: key.Namespace,
+		Name:      depName,
+	}
+	if depErr := c.KubeClient.Get(ctx, depKey, dep); depErr != nil {
+		return fmt.Errorf("failed to run operator: Error getting deployment: %v", depErr)
+	}
+	for _, s := range dep.Status.Conditions {
+		if s.Type == "Available" && s.Status == "True" {
+			return fmt.Errorf("failed to run operator: Deployment is available, but CSV pending")
+		}
+		if s.Type == "Progressing" || s.Type == "ReplicaFailure" {
```

Comment:
Wondering if we should use the constants to do the checks.

https://github.com/kubernetes/api/blob/master/apps/v1/types.go#L418-L430

```golang
if s.Type == appsv1.DeploymentProgressing..
```

@joelanford is there a preference?

Reply:
Prefer constants.

Reply:
+1